sympy/solvers/solveset.py

```diff
@@ -1336,21 +1336,19 @@ def _is_lambert(f, symbol):
 def _solve_as_bivariate(lhs, rhs, symbol, domain):
     poly = lhs.as_poly()
     gens = _filtered_gens(poly, symbol)
+
     if len(gens) == 2:
-        try:
-            gpu = bivariate_type(lhs - rhs, *gens)
-            if gpu is None:
-                raise NotImplementedError
-            g, p, u = gpu
-            inversion = _transolve(g - u, symbol, domain)
-            if inversion:
-                sol = _solveset(p, u, domain)
-                result = FiniteSet(*[i.subs(u, s)
-                                   for i in inversion for s in sol])
+        gpu = bivariate_type(lhs - rhs, *gens)
+        if gpu is None:
+            return
+        g, p, u = gpu
+        inversion = _transolve(g - u, symbol, domain)
```

Comment:
Well, I guess this could cause recursion error. Equation like: `(sin(x)/2 - cos(x)/2)*exp(x) + (sin(x)/2 + cos(x)/2)*exp(x) - 1` causes recursion error.

@smichr Sorry for giving a late update on this PR. I am looking into ways to better handle this scenario. Any suggestions?